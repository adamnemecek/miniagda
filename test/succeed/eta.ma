data P (A : Set) : (A -> A) -> Set
{
  inn : (out : A -> A) -> P A out
}

fun bla : (A : Set) -> (f : (A -> A) -> (A -> A)) ->
  P (A -> A) f ->  P (A -> A) (\ x -> f x)
{
  bla A f p = p    -- (c .(A -> A) f) = c (A -> A) (\ x -> f x)
}
