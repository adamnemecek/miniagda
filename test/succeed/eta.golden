--- opening "eta.ma" ---
--- scope checking ---
--- type checking ---
type  P : ^(A : Set) -> ^ (A -> A) -> Set
term  P.inn : .[A : Set] -> .[out : A -> A] -> < P.inn out : P A out >
term  bla : .[A : Set] -> (f : (A -> A) -> A -> A) -> P (A -> A) f -> P (A -> A) (\ x -> f x)
{ bla [A] f p = p
}
--- evaluating ---
--- closing "eta.ma" ---
